internal/service/networkmanager/vpn_attachment.go

```diff
@@ -0,0 +1,323 @@
+	d.Set("attachment_type", a.AttachmentType)
+	d.Set("core_network_arn", a.CoreNetworkArn)
+	d.Set("core_network_id", a.CoreNetworkId)
+	d.Set("edge_location", a.EdgeLocation)
+	d.Set("owner_account_id", a.OwnerAccountId)
+	d.Set("resource_arn", a.ResourceArn)
+	d.Set("segment_name", a.SegmentName)
+	d.Set("state", a.State)
+	d.Set("vpn_arn", a.ResourceArn)
+
+	tags := KeyValueTags(a.Tags).IgnoreAWS().IgnoreConfig(ignoreTagsConfig)
+
+	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+		return diag.Errorf("Setting tags: %s", err)
+	}
+
+	if err := d.Set("tags_all", tags.Map()); err != nil {
+		return diag.Errorf("setting tags_all: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVpnAttachmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+
+		if err := UpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return diag.FromErr(fmt.Errorf("error updating Network Manager VPN Attachment (%s) tags: %s", d.Get("arn").(string), err))
+		}
+	}
+
+	return resourceVpnAttachmentRead(ctx, d, meta)
+}
+
+func resourceVpnAttachmentDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+
+	// If ResourceAttachmentAccepter is used, then VPN Attachment state
+	// is never updated from StatePendingAttachmentAcceptance and the delete fails
+	output, sErr := FindVpnAttachmentByID(ctx, conn, d.Id())
+	if tfawserr.ErrCodeEquals(sErr, networkmanager.ErrCodeResourceNotFoundException) {
+		return nil
+	}
+
+	if sErr != nil {
+		return diag.Errorf("deleting Network Manager VPN Attachment (%s): %s", d.Id(), sErr)
+	}
+
+	d.Set("state", output.Attachment.State)
+
+	if state := d.Get("state").(string); state == networkmanager.AttachmentStatePendingAttachmentAcceptance || state == networkmanager.AttachmentStatePendingTagAcceptance {
+		return diag.Errorf("cannot delete Network Manager VPN Attachment (%s) in %s state", d.Id(), state)
+	}
+
+	log.Printf("[DEBUG] Deleting Network Manager VPN Attachment: %s", d.Id())
+	_, err := conn.DeleteAttachmentWithContext(ctx, &networkmanager.DeleteAttachmentInput{
+		AttachmentId: aws.String(d.Id()),
+	})
+
+	if tfawserr.ErrCodeEquals(err, networkmanager.ErrCodeResourceNotFoundException) {
+		return nil
+	}
+
+	if err != nil {
+		return diag.Errorf("deleting Network Manager VPN Attachment (%s): %s", d.Id(), err)
+	}
+
+	if _, err := waitVpnAttachmentDeleted(ctx, conn, d.Id(), d.Timeout(schema.TimeoutDelete)); err != nil {
+		return diag.Errorf("waiting for Network Manager VPN Attachment (%s) delete: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func FindVpnAttachmentByID(ctx context.Context, conn *networkmanager.NetworkManager, id string) (*networkmanager.SiteToSiteVpnAttachment, error) {
```

Comment:
More of a general observation, e.g. the ec2 service has a `find.go` for find helper functions. It might make sense to have that in the networkmanager service too

Reply:
I find it easier when the finder functions are located with their corresponding resource